transport_nantes/stripe_app/static/stripe_app/form_flow.js

```diff
@@ -323,18 +323,21 @@ fetch("/donation/config/")
             headers: myHeaders,
             body: form_data,
             method:"POST"})
-        .then((result) => { return result.json(); })
+        .then((result) => {
+            if (!result.ok) {
+                // Stops the redirection to Stripe Checkout if the server
+                // is not responding OK
+                throw new Error("Erreur lors de l'initialisation de la session de paiement")
```

Comment:
What catches this error?
What is the user experience if it happens?

Reply:
On the server's end, it's there : 
https://github.com/transport-nantes/tn_web/blob/3dc1d180a7143675fe2089c3d8a96df6347654fb/transport_nantes/stripe_app/views.py#L211-L215

The 500 is not considered OK (2XX are)

Form validation happens before 

https://github.com/transport-nantes/tn_web/blob/3dc1d180a7143675fe2089c3d8a96df6347654fb/transport_nantes/stripe_app/static/stripe_app/form_flow.js#L232-L239

But let's admit validation did pass, but the problem is stripe API key for example, then hitting submit wouldn't try to redirect you to a checkout session, as none has been created.